addons/binding/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/handler/BoschIndegoHandler.java

```diff
@@ -0,0 +1,252 @@
+        }
+    }
+
+    private boolean isReadyToMow(int statusCode, int error) {
+        // I don´t know why bosch uses different state codes for the same state.
+        return (statusCode == STATE_DOCKED_1 || statusCode == STATE_DOCKED_2 || statusCode == STATE_DOCKED_3
+                || statusCode == STATE_PAUSED || statusCode == STATE_IDLE_IN_LAWN) && error == 0;
+    }
+
+    private boolean verifyCommand(int command, int eshStatus, int statusCode, int errorCode) {
+        // Mower reported an error
+        if (errorCode != 0) {
+            logger.error("The mower reported an error.");
+            return false;
+        }
+        // Command out of range
+        if (command < 1 || command > 3) {
+            logger.debug("Command out of range");
+            return false;
+        }
+        // Command is equal to current state
+        if (command == eshStatus) {
+            logger.debug("Command is equal to state");
+            return false;
+        }
+        // Cant pause while the mower is docked
+        if (command == 3 && eshStatus == 2) {
+            logger.debug("Can´t pause the mower while it´s docked or docking");
+            return false;
+        }
+        // Command means "MOW" but mower is not ready
+        if (command == 1 && !isReadyToMow(statusCode, errorCode)) {
+            logger.debug("The mower is not ready to mow in the moment");
+            return false;
+        }
+        return true;
+    }
+
+    private DeviceCommand getCommandFromEshStatus(int eshStatus) {
+        DeviceCommand command;
+        switch (eshStatus) {
+            case 1:
+                command = DeviceCommand.MOW;
+                break;
+            case 2:
+                command = DeviceCommand.RETURN;
+                break;
+            case 3:
+                command = DeviceCommand.PAUSE;
+                break;
+            default:
+                command = null;
+        }
+        return command;
+    }
+
+    private int getEshStatusFromCommand(DeviceCommand command) {
+        int eshStatus;
+        switch (command) {
+            case MOW:
+                eshStatus = 1;
+                break;
+            case RETURN:
+                eshStatus = 2;
+                break;
+            case PAUSE:
+                eshStatus = 3;
+                break;
+            default:
+                eshStatus = 0;
+        }
+        return eshStatus;
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        logger.debug("removing thing..");
+        running = false;
+        pollingThread.interrupt();
+    }
+
+    @Override
+    public void initialize() {
+        // TODO: Initialize the thing. If done set status to ONLINE to indicate proper working.
+        // Long running initialization should be done asynchronously in background.
+        running = true;
+        pollingThread = new Thread(new Runnable() {
```

Comment:
You must not create your own threads! Please use the scheduler (defined in BaseThingHandler) instead!